### PR TITLE
Total responses data dump handler

### PR DESCRIPTION
**Pull request by cutepig**
- Adds an Athena query that fetches `total_responses` from hardcoded `symptomradar_dev_storage` db
- Pushes the query result to hardcoded bucket `symptomradar-dev-open-data` as `total_responses.json`
- Adds local typings for `athena-express`
- NOTE: Uses `async` functions instead of `Promise`-chains. Discuss.

### CODE REVIEW

File: src/athena-express.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+declare module 'athena-express' {
```
Comment (jareware) [inline src/athena-express.d.ts:1]: :heart_eyes: :clap: :clap:

File: src/backend/main.ts [hunk 2]
```diff
@@ -79,3 +83,23 @@ function hash(input: string, pepper: string) {
     .update(input + pepper)
     .digest('base64');
 }
+
+export async function storeTotalResponsesToS3() {
```
Comment (jareware) [inline src/backend/main.ts:87]: Wrt. `async/await`, I'm fine with either that or naked Promises, but I don't think we should mix the two styles.

Comment (jareware) [reply src/backend/main.ts:87]: That said, my desire for this to end up on `master` might be too great... :wink:

File: src/backend/main.ts [hunk 3]
```diff
@@ -79,3 +83,23 @@ function hash(input: string, pepper: string) {
...
+
+  // TODO: Add model for this
+  const data = queryResult.Items[0];
+
+  await s3
+    .putObject({
+      // TODO: Configure this
+      Bucket: 'symptomradar-dev-open-data',
+      Key: 'total_responses.json',
+      Body: JSON.stringify(data),
```
Comment (jareware) [inline src/backend/main.ts:102]: If you look at the other data sets on https://data.oiretutka.fi/, they're actually pretty-printed JSON.

Not sure if someone finds that distasteful for massive data dumps, but then the JSON gzips to about 10% of its size on the wire anyway.

I would prefer to have this pretty-printed as well.

File: src/index-backend.ts [hunk 4]
```diff
@@ -18,8 +18,15 @@ export const apiEntrypoint: APIGatewayProxyHandler = (event, context) => {
...
 };
 
-export const workerEntrypoint: Handler<unknown> = () => {
+export const workerEntrypoint: Handler<unknown> = async () => {
   console.log('Worker started');
+  try {
+    await storeTotalResponsesToS3();
+    console.log('Worker done');
+  } catch (error) {
+    console.error('ERROR (WORKER)', error);
```
Comment (jareware) [inline src/index-backend.ts:27]: Distinguishing the error source is a nice touch. :ok_hand:

File: src/backend/main.ts (hunk 2 above)
Comment (cutepig) [reply src/backend/main.ts:87]: I have no objections to change this to `Promise.resolve` flow, though personally I've started to favor `async` functions quite highly for a few reasons:

- They clearly indicate that the function is doing some async stuff. I even use the keyword when I'm simply returning a promise, i.e. not even using `await` in the function. I'll refer you to a nice article _What color if your function_ https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/
- Reduces boilerplate, though not much. One handy thing is that it solves the ergonomics for block functions that return a value as `then` handlers, by just getting rid of that paradigm.
- It's already standard in JS, you can use `async` functions in devtools console and Node.js. I also heard rumours that some browsers accept `await` as a top-level statement, very useful in console. Use the platform, Luke.

The only thing that really remains a a small problem in general is wrapping your head around the fact that the code breaks into two whenever you call `await`. The code above it (well up to the previous `await` at least) executes synchronously and the code following `await` is executed at some later point in time. Though this parallels how for e.g. generators work in JS, so it's something that one should just get it's mind set right.

Comment (cutepig) [reply src/backend/main.ts:87]: As a follow-up, I'm willing to convert those other backend functions to `async`/`await` as well ;)

Comment (jareware) [reply src/backend/main.ts:87]: OK, SOLD to the man willing to put his ~money~ patches where his mouth is! :smile: 

Let's merge this in, and do the conversion in a follow-up PR. :+1:

File: infra/modules/main/backend.tf [hunk 5]
```diff
@@ -42,7 +42,8 @@ module "backend_worker" {
   function_s3_bucket     = aws_s3_bucket.backend_code.id
   function_zipfile       = "backend-lambda.zip"
   function_handler       = "index.workerEntrypoint"
-  schedule_expression    = "rate(3 minutes)" # note: full cron expressions are also supported
+  function_timeout       = 60 * 5             # i.e. 5 minutes
```
Comment (cutepig) [inline infra/modules/main/backend.tf:45]: Just curious, how did you settle on these values?

Comment (jareware) [reply infra/modules/main/backend.tf:45]: Pulled them from a hat for now. Let’s revisit when we’re doing more.

**Pull request merged**